### PR TITLE
fix HSA/KFD load for system-wide installation

### PR DESCRIPTION
**Pull request by stalkerg**
It fix for a such issue: #4168
Some distros like Gentoo install HSA and ROCm libs as system-wide libraries instead `/opt` directory. 
ROCM_PATH is more like build-time path, for runtime discovery better to use `find_library` especially if ROCM_PATH not here. 
It's also will work if follow official ROCm instruction https://rocm.docs.amd.com/projects/install-on-linux/en/latest/how-to/native-install/post-install.html

### CODE REVIEW

**Comment by stalkerg** [comment]
ohh you actually generate hsa and compgr files. Hmmm I will fix it.

File: tinygrad/runtime/autogen/comgr.py [hunk 1]
```diff
@@ -27,9 +28,12 @@ def char_pointer_cast(string, encoding='utf-8'):
     return ctypes.cast(string, ctypes.POINTER(ctypes.c_char))
 
 
-
 _libraries = {}
-_libraries['libamd_comgr.so'] = ctypes.CDLL(os.getenv('ROCM_PATH', '/opt/rocm/')+'/lib/libamd_comgr.so')
+_libraries['libamd_comgr.so'] = ctypes.CDLL(
```
Comment (nimlgen) [inline tinygrad/runtime/autogen/comgr.py:32]: Don't change autogenerated files. See `autogen_stubs.sh`

**Comment by geohot** [comment]
We are going to replace HSA with KFD soon anyway.

File: tinygrad/runtime/autogen/comgr.py (hunk 1 above)
Comment (stalkerg) [reply tinygrad/runtime/autogen/comgr.py:32]: thanks, yeah, I found it as well.

**Comment by stalkerg** [comment]
> We are going to replace HSA with KFD soon anyway.

seems like for KFD we still need libamd_comgr.so

**Comment by stalkerg** [comment]
@nimlgen can you check again? Thank you.

**Pull request merged**